docs/proposals/image-updater-first-class-support.md

```diff
@@ -0,0 +1,324 @@
+  - It is harder to validate annotations. Error checking on the controller's end could involve complex string parsing and does not allow us to leverage existing validation solutions provided to us by Kubernetes.
+  - It is harder to express complicated configuration. Leveraging powerful features often results in complex and verbose annotations that are not easy to work with or troubleshoot. 
+It also doesn't store any state, and is, as such, unable to provide some key features that are discussed in further sections. Consolidating this configuration into the Application CR as first class fields in the spec and status is a natural next step in the Image Updater's journey. 
+
+### Goals
+
+- Propose updated Application CR fields to include configuration scoped for image updates
+- Make a case to merge Image Updater controller code into core Argo CD as its own controller 
+- Outline future enhancements planned for Image Updater and its growth within Argo CD 
+
+### Non-Goals
+
+TBD
+
+## Proposal
+
+### Use cases
+
+#### Use case 1:
+As a User of Argo CD, I would like to be able to have my application images updated automatically within expressed constraints when there is a new version available, out of the box with Argo CD. 
+
+#### Use case 2:
+As a user of Argo CD and Argo CD Image Updater, I would like to be able to express advanced Image update configurations through custom resource fields in my application.
+
+For users wanting to use Image Updater in production, it is common to have a strict set of restrictions in place in terms of which images can/should be automatically updated, where these images should come from, which reposoitry/branch the commits should be made to and what credentials should be used among many others. It is important to have a set of first class custom resource fields to be able to express this information accurately and reliably. This would add considerable value to the user experience, including but not limited to increased ease of use, stronger validation and standardization for these fields.
+ 
+#### Use case 3:
+As a user of Argo CD and Argo CD Image Updater, I would like to have image-updater honor rollbacks to my applications by leveraging status information stored in my application.
+
+
+### Image Updater functionality constraints and behaviors
+
+This section highlights some of the key aspects of Image Updater's present design constraints and behaviors in order to provide context for the upcoming sections of the proposal to people who may not be familiar with the project. Full details may be found in the Image Updater documentation (https://argocd-image-updater.readthedocs.io/en/stable/)  
+
+- Image Updater can only update container images for applications whose manifests are rendered using either Kustomize or Helm and - especially in the case of Helm - the templates need to support specifying the image's tag (and possibly name) using a parameter (i.e. image.tag).
```

Comment:
Could it also support setting the image registry via a separate parameter? There are widely used Helm charts in the wild that build the repository name not just from `repository` and `tag` but also `registry`. An example is [prometheus-operator](https://github.com/prometheus-community/helm-charts/blob/3352ef86286a7e8f07266ff3da1a575aa7368c8f/charts/kube-prometheus-stack/templates/prometheus-operator/deployment.yaml#L54).